compiler/rustc_resolve/messages.ftl

```diff
@@ -44,13 +44,32 @@ resolve_binding_shadows_something_unacceptable =
 resolve_binding_shadows_something_unacceptable_suggestion =
     try specify the pattern arguments
 
+resolve_cannot_be_reexported = {$crate_private_reexport ->
+        [true] `{$ident}` is only public within the crate, and cannot be re-exported outside
+        *[false] `{$ident}` is private, and cannot be re-exported
+    }
+
+resolve_cannot_be_reexported_namespace = {resolve_cannot_be_reexported}
+    .label = {$crate_private_reexport ->
+        [true] re-export of crate public `{$ident}`
+        *[false] re-export of private `{$ident}`
```

Comment:
Please don't use pattern matching this way， instead, please using different identifiers and different types.

Reply:
I see, should existing diagnostics written this way be rewritten as well? I used a few as examples when writing this.

What are the valid use-cases for pattern matching? I assume plurals and articles are fine?

Reply:
Yep. Existing diagnostics using pattern matching for different variants of the same message should be rewritten as individual variants.

> What are the valid use-cases for pattern matching? I assume plurals and articles are fine?

Yes, you are exactly right. From fluent's project docs, pattern matching should only be used for _language specific_ variations.